### PR TITLE
control-flow-functions: Update reference manual links

### LINKED ISSUE #16821: Update reference manual links in functions-and-operators/control-flow-functions.md

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #16821

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/control-flow-functions.md [hunk 1]
```diff
@@ -6,11 +6,115 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
++------+----------------------------+
+| NULL | x has no value             |
+|    1 | 1                          |
++------+----------------------------+
+2 rows in set (0.0006 sec)
+```
+
+## NULLIF()
+
+The `NULLIF()` function returns null if both arguments are the same. Otherwise it returns the first argument.
```
Comment (mjonss) [inline functions-and-operators/control-flow-functions.md:95]: Can we clarify what happens when one or more of the arguments are NULL?

**Review by mjonss — APPROVED** [review]
LGTM, with a minor suggestion.

File: functions-and-operators/control-flow-functions.md [hunk 2]
```diff
@@ -6,11 +6,114 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
+|  9 | odd                        |
+| 10 | even                       |
++----+----------------------------+
+10 rows in set (0.00 sec)
+```
+
+## IFNULL()
+
+The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data
+of the second argument of the function.
```
Comment (dveeden) [inline functions-and-operators/control-flow-functions.md:75]:
```suggestion
The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data of the second argument of the function.
```

File: functions-and-operators/control-flow-functions.md (hunk 1 above)
Comment (dveeden) [reply functions-and-operators/control-flow-functions.md]:
```
mysql> SELECT NULLIF(1, 2);
+--------------+
| NULLIF(1, 2) |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, 1);
+--------------+
| NULLIF(1, 1) |
+--------------+
|         NULL |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, NULL);
+-----------------+
| NULLIF(1, NULL) |
+-----------------+
|               1 |
+-----------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, NULL);
+--------------------+
| NULLIF(NULL, NULL) |
+--------------------+
|               NULL |
+--------------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, 1);
+-----------------+
| NULLIF(NULL, 1) |
+-----------------+
|            NULL |
+-----------------+
1 row in set (0.00 sec)
```

**Review by bb7133 — APPROVED** [review]
LGTM

File: functions-and-operators/control-flow-functions.md [hunk 3]
```diff
@@ -6,11 +6,113 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
 | [`IFNULL()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) | Null if/else construct            |
 | [`NULLIF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) | Return NULL if expr1 = expr2      |
+
+## CASE
+
+The `CASE` operator allows you to respond on multiple values and conditions.
+
+```sql
+WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
+SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d
```
Comment (qiancai) [inline functions-and-operators/control-flow-functions.md:24]:
```suggestion
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d;
```

File: functions-and-operators/control-flow-functions.md [hunk 4]
```diff
@@ -6,11 +6,113 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
+|  6 |               12 |
+|  7 |               14 |
+|  8 |               16 |
+|  9 |               18 |
+| 10 |               20 |
++----+------------------+
+10 rows in set (0.00 sec)
+```
+
+Here n+n and n+2 both evaluate to 4 if n is 2, which makes both arguments the same and the function return NULL.
```
Comment (qiancai) [inline functions-and-operators/control-flow-functions.md:118]:
```suggestion
In this example, when `n` equals `2`, both `n+n` and `n+2` equal to `4`, making both arguments the same and causing the function to return `NULL`.
```

File: functions-and-operators/control-flow-functions.md [hunk 5]
```diff
@@ -6,11 +6,141 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
 # Control Flow Functions
 
-TiDB supports all of the [control flow functions](https://dev.mysql.com/doc/refman/5.7/en/flow-control-functions.html) available in MySQL 5.7.
+TiDB supports all of the [control flow functions](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html) available in MySQL 8.0.
 
 | Name                                                                                            | Description                       |
 |:--------------------------------------------------------------------------------------------------|:----------------------------------|
-| [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case)       | Case operator                     |
+| [`CASE`](#case)       | Case operator                     |
 | [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if)         | If/else construct                 |
```
Comment (qiancai) [inline functions-and-operators/control-flow-functions.md:14]:
```suggestion
| [`IF()`](#if)         | If/else construct                 |
```

File: functions-and-operators/control-flow-functions.md [hunk 6]
```diff
@@ -6,11 +6,141 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
-| [`IFNULL()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) | Null if/else construct            |
-| [`NULLIF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) | Return NULL if expr1 = expr2      |
+| [`IFNULL()`](#if) | Null if/else construct            |
```
Comment (qiancai) [inline functions-and-operators/control-flow-functions.md:15]:
```suggestion
| [`IFNULL()`](#ifnull) | Null if/else construct            |
```

File: functions-and-operators/control-flow-functions.md [hunk 7]
```diff
@@ -6,11 +6,141 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
+|  6 |               12 |
+|  7 |               14 |
+|  8 |               16 |
+|  9 |               18 |
+| 10 |               20 |
++----+------------------+
+10 rows in set (0.00 sec)
+```
+
+In this example, when `n` equals `2`, both `n+n` and `n+2` equal to `4`, making both arguments the same and causing the function to return `NULL`.
```
Comment (qiancai) [inline functions-and-operators/control-flow-functions.md:146]:
```suggestion
In this example, when `n` equals `2`, both `n+n` and `n+2` equal `4`, making both arguments the same and causing the function to return `NULL`.
```

**Review by qiancai — APPROVED** [review]

File: functions-and-operators/control-flow-functions.md [hunk 8]
```diff
@@ -6,11 +6,138 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
 | Name                                                                                            | Description                       |
 |:--------------------------------------------------------------------------------------------------|:----------------------------------|
-| [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case)       | Case operator                     |
-| [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if)         | If/else construct                 |
-| [`IFNULL()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) | Null if/else construct            |
-| [`NULLIF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) | Return NULL if expr1 = expr2      |
+| [`CASE`](#case)       | Case operator                     |
+| [`IF()`](#if)         | If/else construct                 |
+| [`IFNULL()`](#ifnull) | Null if/else construct            |
+| [`NULLIF()`](#nullif) | Return NULL if expr1 = expr2      |
```
Comment (Oreoxmt) [inline functions-and-operators/control-flow-functions.md:16]:
```suggestion
| [`NULLIF()`](#nullif) | Return `NULL` if expr1 = expr2      |
```

File: functions-and-operators/control-flow-functions.md [hunk 9]
```diff
@@ -6,11 +6,138 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
+
+## CASE
+
+The [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case) operator allows you to perform conditional logic and customize query results based on specified conditions.
```
Comment (Oreoxmt) [inline functions-and-operators/control-flow-functions.md:20]:
```suggestion
The [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case) operator enables you to perform conditional logic and customize query results based on specified conditions.
```

File: functions-and-operators/control-flow-functions.md [hunk 10]
```diff
@@ -6,11 +6,138 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
...
+|  8 | even                                         |
+|  9 | odd                                          |
+| 10 | even                                         |
++----+----------------------------------------------+
+10 rows in set (0.00 sec)
+```
+
+## IF()
+
+The [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if) function allows you to perform different actions based on whether a value or expression is true or not.
```
Comment (Oreoxmt) [inline functions-and-operators/control-flow-functions.md:60]:
```suggestion
The [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if) function enables you to perform different actions based on whether a value or expression is true or not.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**